### PR TITLE
[CDAP-14868] Deterministic behavior for batch of non-disjoint metadata mutations

### PR DESCRIPTION
**Pull request by anew**
If a batch of mutations contains duplicate entity ids, then don't perform them in parallel, but in sequence. Otherwise they would overwrite each other's changes.

Also make sure that an empty batch of mutations does not cause errors (Elasticsearch throws an exception for empty multi-get).

### CODE REVIEW

**Comment by anew** [comment]
Builds passed:
DUT: https://builds.cask.co/browse/CDAP-DUT6863
DBV: https://builds.cask.co/browse/CDAP-DBV22
ITM: https://builds.cask.co/browse/IT-ITM-122

File: cdap-elastic/src/main/java/co/cask/cdap/metadata/elastic/ElasticsearchMetadataStorage.java [hunk 1]
```diff
@@ -319,10 +320,27 @@ public MetadataChange apply(MetadataMutation mutation) throws IOException {
 
   @Override
   public List<MetadataChange> batch(List<? extends MetadataMutation> mutations) throws IOException {
+    if (mutations.isEmpty()) {
+      return Collections.emptyList();
+    }
+    Set<MetadataEntity> entities = new HashSet<>();
```
Comment (poornachandra) [inline cdap-elastic/src/main/java/co/cask/cdap/metadata/elastic/ElasticsearchMetadataStorage.java:326]: Is it possible to use a Map< MetadataEntity, List<MetadataMutation>> to group together the mutations for each entity, and then execute the disjoint mutations concurrently?

**Review by poornachandra — COMMENTED** [review]
@anew Just one question

File: cdap-elastic/src/main/java/co/cask/cdap/metadata/elastic/ElasticsearchMetadataStorage.java (hunk 1 above)
Comment (anew) [reply cdap-elastic/src/main/java/co/cask/cdap/metadata/elastic/ElasticsearchMetadataStorage.java:326]: yes, it is but I did not think it is worth the complexity. Do you think it should be done? I opened a Jira here: https://issues.cask.co/browse/CDAP-14979

Comment (poornachandra) [reply cdap-elastic/src/main/java/co/cask/cdap/metadata/elastic/ElasticsearchMetadataStorage.java:326]: 👍

**Review by poornachandra — APPROVED** [review]

**Pull request merged**